bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.kt

```diff
@@ -0,0 +1,164 @@
+        apiLevel: Int, is32bit: Boolean,
+        appVersion: String,
+        buildUuid: String,
+        releaseStage: String
+    )
+
+    external fun startedSession(
+        sessionID: String,
+        key: String,
+        handledCount: Int,
+        unhandledCount: Int
+    )
+
+    external fun deliverReportAtPath(filePath: String)
+    external fun addBreadcrumb(name: String, type: String, timestamp: String, metadata: Any)
+    external fun addMetadataString(tab: String, key: String, value: String)
+    external fun addMetadataDouble(tab: String, key: String, value: Double)
+    external fun addMetadataBoolean(tab: String, key: String, value: Boolean)
+    external fun addHandledEvent()
+    external fun addUnhandledEvent()
+    external fun clearBreadcrumbs()
+    external fun clearMetadataTab(tab: String)
+    external fun removeMetadata(tab: String, key: String)
+    external fun pausedSession()
+    external fun updateContext(context: String)
+    external fun updateInForeground(inForeground: Boolean, activityName: String)
+    external fun updateOrientation(orientation: Int)
+    external fun updateUserId(newValue: String)
+    external fun updateUserEmail(newValue: String)
+    external fun updateUserName(newValue: String)
+
+    /**
+     * Creates a new native bridge for interacting with native components.
+     * Configures logging and ensures that the reporting directory exists
+     * immediately.
+     */
+    init {
+        val outFile = File(reportDirectory)
+        if (!outFile.exists() && !outFile.mkdirs()) {
+            warn("The native reporting directory cannot be created.")
+        }
+    }
+
+    override fun update(observable: Observable, arg: Any?) {
+        if (isInvalidMessage(arg)) return
+
+        when (val msg = arg as StateEvent) {
+            is Install -> handleInstallMessage(msg)
+            DeliverPending -> deliverPendingReports()
+            is AddMetadata -> handleAddMetadata(msg)
+            ClearBreadcrumbs -> clearBreadcrumbs()
+            is ClearMetadataTab -> clearMetadataTab(msg.section)
+            is RemoveMetadata -> removeMetadata(msg.section, msg.key ?: "")
+            is AddBreadcrumb -> addBreadcrumb(msg.message, msg.type.toString(), msg.timestamp, msg.metadata)
+            NotifyHandled -> addHandledEvent()
+            NotifyUnhandled -> addUnhandledEvent()
+            PauseSession -> pausedSession()
+            is StartSession -> startedSession(msg.id, msg.startedAt, msg.handledCount, msg.unhandledCount)
+            is UpdateContext -> updateContext(msg.context ?: "")
+            is UpdateInForeground -> updateInForeground(msg.inForeground, msg.contextActivity ?: "")
+            is UpdateOrientation -> updateOrientation(msg.orientation)
+            is UpdateUserEmail -> updateUserEmail(msg.email ?: "")
+            is UpdateUserName -> updateUserName(msg.name ?: "")
+            is UpdateUserId -> updateUserId(msg.id ?: "")
+        }
```

Comment:
Do we need any sort of warning if we reach the default case? We didn't before, but it seemed like a simpler mapping between the `MessageType` enum values and the switch statement.

Reply:
A default case is not required in Kotlin when using a sealed class in a `when` expression, as the compiler will verify that all cases are covered: https://kotlinlang.org/docs/reference/sealed-classes.html